UnitTests/Plugins/BuildServerIntegration/AppVeyorIntegrationTests/AppVeyorAdapterTests.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Linq;
+using AppVeyorIntegration;
+using FluentAssertions;
+using GitUIPluginInterfaces;
+using GitUIPluginInterfaces.BuildServerIntegration;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace AppVeyorIntegrationTests
+{
+    [TestFixture]
+    public class AppVeyorAdapterTests
+    {
+        private AppVeyorAdapter.Project _project = new AppVeyorAdapter.Project
+            { Id = "ProjectId", Name = "ProjectName", QueryUrl = "ProjectQueryUrl" };
+
+        [Test]
+        public void Should_return_no_build_Info_When_Api_Json_is_empty()
+        {
+            var buildInfo = new AppVeyorAdapter().ExtractBuildInfo(_project, string.Empty);
+
+            buildInfo.Should().HaveCount(0);
+        }
+
+        private const string ContentBegin = @"{
```

Comment:
Let's move large chunks of input data into physical files and assert the result with ApprovalTests.
Load:
https://github.com/gitextensions/gitextensions/blob/d0e8c63c666b6d81630b78d5a9b378a2de64b8f4/UnitTests/GitCommandsTests/UserRepositoryHistory/Legacy/RepositoryCategoryXmlSerialiserTests.cs#L33-L39

Assert (few overloads available):
https://github.com/gitextensions/gitextensions/blob/d0e8c63c666b6d81630b78d5a9b378a2de64b8f4/UnitTests/GitCommandsTests/UserRepositoryHistory/Legacy/RepositoryCategoryXmlSerialiserTests.cs#L105
This will create a `testname.approved.txt` file that can be viewed in a normal diff viewer. The *.approved.* files gets committed into the repo.

This will significantly simplify the maintenance of tests and change of data, if necessary.